src/main/java/com/flightstats/hub/dao/aws/DynamoWebhookDao.java

```diff
@@ -69,31 +68,34 @@ public void upsert(Webhook webhook) {
 
     @Override
     public Webhook get(String name) {
+        return getDynamoItem(name)
+                .flatMap(this::mapItem)
+                .orElse(null);
+    }
+
+    private Optional<Map<String, AttributeValue>> getDynamoItem(String name) {
         HashMap<String, AttributeValue> keyMap = new HashMap<>();
         keyMap.put("name", new AttributeValue(name));
         try {
             GetItemResult result = dbClient.getItem(dynamoProperties.getWebhookConfigTableName(), keyMap, true);
-            if (result.getItem() == null) {
-                return null;
-            }
-            return tryMapItem(result.getItem())
-                    .orElseThrow(() -> new ResourceNotFoundException("Unable to read webhook config from dynamo " + name));
+            return Optional.ofNullable(result)
+                    .map(GetItemResult::getItem);
         } catch (ResourceNotFoundException e) {
```

Comment:
Hmm. With this change the problem shifted to next step :) Exception block is never hit when getItem is null. Optional.ofNullable returns Optional.empty when item is null.

Reply:
Had a offline convo with Laurie. The ResourceNotFoundException is for the dbClient.getItem call. This makes sense now.

Reply:
Talked to Manjula about this. Like 80 throws a `ResourceNotFoundException` during `getItem`. Then the AWS SDK API is super helpful by adding a second `getItem` method to get the map of record attributes, which...apparently might be null.